pilot/pkg/serviceregistry/kube/conversion.go

```diff
@@ -184,12 +184,6 @@ func kubeToIstioServiceAccount(saname string, ns string) string {
 
 // SecureNamingSAN creates the secure naming used for SAN verification from pod metadata
 func SecureNamingSAN(pod *coreV1.Pod) string {
-
-	//use the identity annotation
-	if identity, exist := pod.Annotations[annotation.AlphaIdentity.Name]; exist {
```

Comment:
Why are we removing this ?

Reply:
is this still used? I talked with @incfly who mentioned it is likely a legacy (Istio 0.2) artifact. The reason I am touching it here is because if we don't then we will need to make the code more flexible to support custom identities. If we need it, I can refactor a bit - not a big deal

Reply:
😞 It's sad this undocumented "feature" was removed. It was _really_ useful for users who brought their own mTLS certificates. I used a mutating webhook to add this pod annotation dynamically, rather than hard code the cert's SPIFFE SAN into each service's destination rule by hand.

Reply:
Yeah - currently workload proxies hardcode SAN validation in their envoy configs based on what they assume the SAN will be - if you happen to use a different SDS or cert generation mechanism than the default, the assumption doesn't match the actual cert supplied to the workload.

As long as this is the case, having a way to work around this other than creating destinationrules for every service would be something I'd prefer to keep around, as @rmichela says.

Reply:
I see why @howardjohn removed it - some things depend on a specific SPIFFE ID format so arbitrary ones are a problem - but I think that points to a deeper issue where the code treats SANs, serviceaccounts, and SPIFFE IDs as interchangeable concepts with the same meaning when they aren't.

Reply:
It's safe to assume the SPIFFE ID is in the SAN (that's just SVID), but it seems like a bad idea to assume anything about the contents of the SPIFFE path. The path can be hierarchical, key/value pair, or completely opaque. https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#22-path